plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.test.ts

```diff
@@ -542,4 +671,368 @@ describe('BitbucketServerEntityProvider', () => {
+  });
+
+  it('add onRepoPush', async () => {
+    const schedule = new PersistingTaskRunner();
+    setupRepositoryReqHandler('master');
+    authService.getPluginRequestToken.mockResolvedValue({
+      token: 'fake-token',
+    });
+    const addedModule = createLocationEntity(
+      test1RepoUrl,
+      `/added-module:${targetPath}`,
+      'master',
+    );
+
+    const entityProviderConnection: EntityProviderConnection = {
+      applyMutation: jest.fn(),
+      refresh: jest.fn(),
+    };
+
+    const config = new ConfigReader({
+      integrations: {
+        bitbucketServer: [
+          {
+            host: host,
+          },
+        ],
+      },
+      catalog: {
+        providers: {
+          bitbucketServer: {
+            mainProvider: {
+              host: host,
+              apiBaseUrl: `https://${host}/rest/api/1.0`,
+              catalogPath: `/added-module:/catalog-info.yaml`,
+            },
+          },
+        },
+      },
+    });
+
+    const catalogApi = {
+      getEntities: async (
+        _request: { filter: Record<string, string> },
+        _options: { token: string },
+      ): Promise<{ items: Entity[] }> => {
+        return {
+          items: [],
+        };
+      },
+    };
+    const provider = BitbucketServerEntityProvider.fromConfig(config, {
+      catalogApi: catalogApi as any as CatalogApi,
+      logger,
+      schedule,
+      events,
+      auth: authService as any as AuthService,
+    })[0];
+
+    await provider.connect(entityProviderConnection);
+    await events.publish(repoPushEventParams);
+    const addedEntities = [
+      {
+        entity: addedModule,
+        locationKey: 'bitbucketServer-provider:mainProvider',
+      },
+    ];
+    const removedEntities: DeferredEntity[] = [];
+
+    expect(entityProviderConnection.refresh).toHaveBeenCalledTimes(1);
+    expect(entityProviderConnection.refresh).toHaveBeenCalledWith({
+      keys: [],
+    });
+    expect(entityProviderConnection.applyMutation).toHaveBeenCalledTimes(1);
+    expect(entityProviderConnection.applyMutation).toHaveBeenCalledWith({
+      type: 'delta',
+      added: addedEntities,
+      removed: removedEntities,
+    });
+  });
+
+  it('fail add onRepoPush from wrong default branch', async () => {
+    const schedule = new PersistingTaskRunner();
+    setupRepositoryReqHandler('main');
+
+    const entityProviderConnection: EntityProviderConnection = {
+      applyMutation: jest.fn(),
+      refresh: jest.fn(),
+    };
+
+    const config = new ConfigReader({
+      integrations: {
+        bitbucketServer: [
+          {
+            host: host,
+          },
+        ],
+      },
+      catalog: {
+        providers: {
+          bitbucketServer: {
+            mainProvider: {
+              host: host,
+              apiBaseUrl: `https://${host}/rest/api/1.0`,
+              catalogPath: `/added-module:/catalog-info.yaml`,
+            },
+          },
+        },
+      },
+    });
+
+    const catalogApi = {
```

Comment:
when using the CatalogService type, you have test utilities to base your tests on - check out the `catalogServiceMock` ([example here](https://github.com/backstage/backstage/blob/f1ac148efdb7d23319e1ab90154dda43e5a4e0c3/plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.test.ts#L108)) which actually functions like a catalog so you don't have to fake individual methods! And you don't have to do the awkward `as any as CatalogApi` cast below either since it's all of the right types all the way. There IS a `catalogServiceMock.mock({ .. })` as well if you absolutely have to.